internal/kms/kms/gcp/gcp.go

```diff
@@ -8,328 +8,63 @@ SPDX-License-Identifier: AGPL-3.0-only
 Package gcp implements a KMS backend for Google Cloud KMS.
 
 The following permissions are required for the service account used to authenticate with GCP:
-
-  - cloudkms.cryptoKeyVersions.create
-
-  - cloudkms.cryptoKeyVersions.update
-
-  - cloudkms.cryptoKeyVersions.useToDecrypt
-
-  - cloudkms.cryptoKeyVersions.useToEncrypt
-
-  - cloudkms.importJobs.create
-
-  - cloudkms.importJobs.get
-
-  - cloudkms.importJobs.useToImport
+  - cloudkms.cryptoKeys.get
+  - cloudkms.cryptoKeys.encrypt
+  - cloudkms.cryptoKeys.decrypt
 */
 package gcp
 
 import (
 	"context"
-	"crypto/rand"
-	"crypto/rsa"
-	"crypto/sha1"
-	"errors"
 	"fmt"
 	"io"
-	"strings"
-	"time"
 
-	kms "cloud.google.com/go/kms/apiv1"
-	"cloud.google.com/go/kms/apiv1/kmspb"
-	"github.com/edgelesssys/constellation/v2/internal/kms/config"
 	kmsInterface "github.com/edgelesssys/constellation/v2/internal/kms/kms"
-	"github.com/edgelesssys/constellation/v2/internal/kms/kms/util"
+	"github.com/edgelesssys/constellation/v2/internal/kms/kms/internal"
 	"github.com/edgelesssys/constellation/v2/internal/kms/storage"
-	"github.com/googleapis/gax-go/v2"
-	"google.golang.org/api/option"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
+	"github.com/edgelesssys/constellation/v2/internal/kms/uri"
+	gcpckms "github.com/hashicorp/go-kms-wrapping/wrappers/gcpckms/v2"
 )
 
-type clientAPI interface {
-	io.Closer
-	CreateCryptoKey(context.Context, *kmspb.CreateCryptoKeyRequest, ...gax.CallOption) (*kmspb.CryptoKey, error)
-	CreateImportJob(context.Context, *kmspb.CreateImportJobRequest, ...gax.CallOption) (*kmspb.ImportJob, error)
-	Decrypt(context.Context, *kmspb.DecryptRequest, ...gax.CallOption) (*kmspb.DecryptResponse, error)
-	Encrypt(context.Context, *kmspb.EncryptRequest, ...gax.CallOption) (*kmspb.EncryptResponse, error)
-	GetKeyRing(context.Context, *kmspb.GetKeyRingRequest, ...gax.CallOption) (*kmspb.KeyRing, error)
-	ImportCryptoKeyVersion(context.Context, *kmspb.ImportCryptoKeyVersionRequest, ...gax.CallOption) (*kmspb.CryptoKeyVersion, error)
-	UpdateCryptoKeyPrimaryVersion(context.Context, *kmspb.UpdateCryptoKeyPrimaryVersionRequest, ...gax.CallOption) (*kmspb.CryptoKey, error)
-	GetImportJob(context.Context, *kmspb.GetImportJobRequest, ...gax.CallOption) (*kmspb.ImportJob, error)
-}
-
 // KMSClient implements the CloudKMS interface for Google Cloud Platform.
 type KMSClient struct {
-	projectID        string
-	locationID       string
-	keyRingID        string
-	newClient        func(ctx context.Context, opts ...option.ClientOption) (clientAPI, error)
-	waitBackoffLimit int
-	storage          kmsInterface.Storage
-	protectionLevel  kmspb.ProtectionLevel
-	kekID            string
-	opts             []gax.CallOption
+	kms    *internal.KMSClient
+	client io.Closer
```

Comment:
Why have this separate property? Couldn't the Close be defined on the KMSClient directly? Especially since it modifies the `kms` property.

Reply:
`Close` in this case, is a method on the GCP API client used by the kms wrapper. The wrapper interface itself does not have a `Close` method
For `Close` to be a method of the internal KMS Client we would need to run a type assertion on the underlying API client of the wrapper and then call `Close` on it if it has such a method.

Not sure if this would be cleaner